packages/chats/src/BubbleMedia/MediaImage.tsx

```diff
@@ -1,153 +1,62 @@
-    const { onError, onLoad } = message;
+    const imageRef = React.useRef<Image>(null);
+    const { onError, onLoad, preview, data, onLayout } = message;
     const containerStyle = useBubbleContainerStyle(message);
     const bubbleBackgroundColor = useBubbleBackgroundColor(message);
     const styles = useStyles();
-    const [
-        imageOriginalSize,
-        setImageOriginalSize,
-    ] = React.useState<ImageSize | null>(null);
-    const [imageSize, setImageSize] = React.useState<ImageSize | null>(null);
-    const windowWidth = useWindowDimensions().width;
 
-    const maxImageSize = useMaxImageSize(windowWidth);
+    const imageSize = useImageSize(data);
 
-    React.useEffect(() => {
-        if (imageOriginalSize) {
-            const newImageSize = getImageSize(
-                imageOriginalSize?.width,
-                imageOriginalSize?.height,
-                maxImageSize.width,
-                maxImageSize.height,
-            );
-            setImageSize(newImageSize);
-        }
-    }, [
-        imageOriginalSize,
-        imageOriginalSize?.width,
-        imageOriginalSize?.height,
-        maxImageSize.width,
-        maxImageSize.height,
-    ]);
+    const { onErrorCallback, onLoadCallback } = useImageCallback(data, onError, onLoad);
 
-    React.useEffect(() => {
-        if (message.data) {
-            Image.getSize(message.data, (width, height) =>
-                setImageOriginalSize({
-                    width,
-                    height,
-                }),
-            );
-        }
-    }, [message.data]);
+    const [isOpen, setIsOpen] = React.useState<boolean>(false);
 
-    const { onErrorCallback, onLoadCallback } = useImageCallback(
-        message.data,
-        onError,
-        onLoad,
+    const { fullSizeImage, previewImage } = useImages(
+        data,
+        preview,
+        imageRef,
+        imageSize,
+        onErrorCallback,
+        onLoadCallback,
     );
 
-    if (!message.data) {
+    const onPress = React.useCallback(() => {
+        setIsOpen(prevIsOpen => !prevIsOpen);
+    }, []);
+
+    const onClose = React.useCallback(() => {
+        setIsOpen(false);
+    }, []);
+
+    if (!previewImage || !fullSizeImage) {
         return null;
     }
+
     return (
-        <View
-            style={[containerStyle, styles.container]}
-            onLayout={message.onLayout}
-        >
-            <View style={[bubbleBackgroundColor, styles.bubble]}>
-                <UIImage
-                    source={{ uri: message.data }}
-                    style={imageSize}
-                    onError={onErrorCallback}
-                    onLoad={onLoadCallback}
-                />
-            </View>
+        <View style={[containerStyle, styles.container]} onLayout={onLayout}>
+            <TouchableOpacity activeOpacity={1} onPress={onPress}>
+                <View style={[bubbleBackgroundColor, styles.bubble]}>
```

Comment:
Maybe apply the styles to upper `touchable` to not create additional `View`?

Reply:
I tried. Styles doesn't work with TouchableWithoutFeedback

Reply:
Looks like [`TouchableOpacity`](https://reactnative.dev/docs/0.64/touchableopacity#style) has it :)